src/init.cpp

```diff
@@ -208,6 +209,7 @@ void DynamicDataLoader::initialize()
     add( "gate", &gates::load_gates );
     add( "overlay_order", &load_overlay_ordering );
     add( "mission_definition", []( JsonObject &jo ) { mission_type::load_mission_type( jo ); } );
+    add( "weather_settings", &weather_generator::load );
```

Comment:
The static `load` function loads and returns the object, but it's not stored anywhere. And there is no object with `"type": "weather_settings"` in any JSON file, so this will (currently) not be triggered anyway. The code for the regional settings loads the generator explicitly, without using the `DynamicDataLoader`.